cookbooks/equilibrium_grain_size/plugins/equilibrium_grain_size.h

```diff
@@ -0,0 +1,497 @@
+                                      const Point<dim> &position,
+                                      const double      viscosity_guess = 0) const;
+
+        /**
+         * This function calculates the dislocation viscosity for a given
+         * dislocation strain rate.
+         */
+        double dislocation_viscosity_fixed_strain_rate (const double      temperature,
+                                                        const double      pressure,
+                                                        const double      second_strain_rate_invariant,
+                                                        const Point<dim> &position) const;
+
+        double density (const double temperature,
+                        const double pressure,
+                        const std::vector<double> &compositional_fields,
+                        const Point<dim> &position) const;
+
+        double compressibility (const double temperature,
+                                const double pressure,
+                                const std::vector<double> &compositional_fields,
+                                const Point<dim> &position) const;
+
+        double thermal_expansivity (const double temperature,
+                                    const double pressure,
+                                    const std::vector<double> &compositional_fields,
+                                    const Point<dim> &position) const;
+
+        /**
+         * Returns the p-wave velocity as calculated by HeFESTo.
+         */
+        double seismic_Vp (const double      temperature,
+                           const double      pressure,
+                           const std::vector<double> &compositional_fields,
+                           const Point<dim> &position) const;
+
+        /**
+         * Returns the s-wave velocity as calculated by HeFESTo.
+         */
+        double seismic_Vs (const double      temperature,
+                           const double      pressure,
+                           const std::vector<double> &compositional_fields,
+                           const Point<dim> &position) const;
+
+        /**
+         * Function that defines the phase transition interface
+         * (0 above, 1 below the phase transition).This is done
+         * individually for each transition and summed up in the end.
+         */
+        double
+        phase_function (const Point<dim> &position,
+                        const double temperature,
+                        const double pressure,
+                        const unsigned int phase) const;
+
+        /**
+         * Function that returns the phase for a given
+         * position, temperature, pressure and compositional
+         * field index.
+         */
+        unsigned int
+        get_phase_index (const Point<dim> &position,
+                         const double temperature,
+                         const double pressure) const;
```

Comment:
Is the documentation for these two functions switched? The documentation for the first function describes a sum over all phases but the function takes the `phase` as an argument. The second function does not take such an argument, so one may assume that it sums over all phases?

Reply:
So these functions seem to exist in identical form in the grain_size material model (with the same documentation). 
I think the documentation is not switched, just outdates. But if we change it here, it will still be wrong in the grain size model. Maybe it would make sense to open and issue to unify these functions later?

The `get_phase_index` computes the index of the stable phase for a given position, temperature and pressure. 

The `phase_function` computes the value of the phase function (which is the fraction of material that has already undergone a certain phase transition) for an individual transition. 
(So there's no summing up as part of the function.)